go-controller/pkg/util/pod_annotation.go

```diff
@@ -570,6 +590,19 @@ func AddRoutesGatewayIP(
 					podAnnotation.Gateways = append(podAnnotation.Gateways, gatewayIPnet.IP)
 				}
 			}
+			// Until https://github.com/ovn-kubernetes/ovn-kubernetes/issues/4876 is fixed, it is limited to IC only
+			if config.OVNKubernetesFeature.EnableInterconnect {
+				if _, isIPv6Mode := netinfo.IPMode(); isIPv6Mode {
+					joinAddrs, err := ParseNodeGatewayRouterJoinAddrs(node, netinfo.GetNetworkName())
+					if err != nil {
+						return err
+					}
+					if len(joinAddrs) == 0 {
+						return nil
```

Comment:
I think we should return error.

If we don't, the GW will not be properly oconfigured, thus giving the user something that is not what they've requested.

Reply:
done, yep also raised it here https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4878#discussion_r1895789390